File: samples/server/petstore/finch/src/main/scala/ApiResponse.scala
package io.swagger.petstore

import argonaut.Argonaut._
import argonaut.CodecJson

/**
 * Describes the result of uploading an image resource
 * @param code 
 * @param _type 
 * @param message 
 */
case class ApiResponse(code: Option[Int],
                _type: Option[String],
                message: Option[String]
                )

object ApiResponse {
  /**
   * Creates the codec for converting ApiResponse from and to JSON.
   */
  implicit val ApiResponseCodec: CodecJson[ApiResponse] = casecodec3(ApiResponse.apply, ApiResponse.unapply)("code", "type", "message")
}

File: samples/client/petstore/flutter/swagger/lib/model/api_response.dart
part of swagger.api;

class ApiResponse {
  
  int code = null;
  

  String type = null;
  

  String message = null;
  
  ApiResponse();

  @override
  String toString() {
    return 'ApiResponse[code=$code, type=$type, message=$message, ]';
  }

  ApiResponse.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    code = json['code'];
    type = json['type'];
    message = json['message'];
  }

  Map<String, dynamic> toMap() {
    return {
      'code': code,
      'type': type,
      'message': message
     };
  }
}

